### ISSUE TITLE
beforeInject not called on relationships (1.0.0-beta.1)

### ISSUE THREAD

**Issue by astrism** [js-data/js-data#35]
I have hasMany relationship on my Profile object, when Profile.find returns and automatically injects the Profile it also injects the hasMany relationship for Suggestions. Unfortunately it doesn't call beforeInject as it used to before I switched over to js-data. When I step through the js-data code I noticed something odd. definition.beforeInject is defined correctly, but right after DSUtils._ is called, options.beforeInject is set to default method rather than my override.

```
  var name = definition.name;
  options = DSUtils._(definition, options);


  if (options.notify) {
    options.beforeInject(options, attrs);
```
 

You can see the definition for both below:
Profile:
```
var definition = {
	name: 'profile',
	defaultAdapter: 'profileAdapter',
	relations: {
		hasOne: {
			user: {
				localField: 'owner',
				foreignKey: 'id'
			}
		},
		hasMany: {
			suggestion: {
				localField: 'suggestions',
				foreignKey: 'ownerId'
			}
		}
	},
	computed: {
		id: ['owner', _updateId]
	},
	methods: {
		// Instance methods
		updateLinks: _updateLinks
	}
}
```

Suggestion:
```
var definition = {
	name: 'suggestion',
	defaultAdapter: 'suggestionAdapter',
	relations: {
		belongsTo: {
			user: {
				localField: 'owner',
				localKey: 'ownerId'
			}
		}
	},
	computed: {
		ownerId: ['owner', _updateOwnerId],
		votes: ['totalVotes', ParseData.defaultValueHandler(0)],
		skips: ['skipped', ParseData.defaultValueHandler(0)],
		views: ['votes', 'skips', _updateViews],
		slug: ['text', _updateSlug],
		link: ['slug', 'id', _updateLink],
		kdr: ['votes', 'skips', _updateKdr],
		typeClass: ['type', _updateTypeClass],
		typeDisplay: ['type', _updateTypeDisplay],
		url: ['url', _updateUrl],
		imageUrl: ['type', _updateImageUrl]
	},
	methods: {
		// Instance methods
		updateLinks: _updateLinks
	},
	beforeInject: _beforeInject,
	afterInject: _afterInject
}
```
I'll see if I can put together a plnkr to illustrate this.

**Comment by kentcdodds** [comment]
@astrism fancy meeting you here :-)

**Comment by astrism** [comment]
I tend to break everything, it was bound to happen.

**Comment by jmdobry** [comment]
Your `beforeInject` handler should be in the prototype chain of the `options` object and should be the function that is invoked if you didn't pass in a handler in the options. A plnkr that demonstrates a bug in this would be most helpful.

**Comment by jmdobry** [comment]
This might be related to #42, which I just fixed in [1.2.1](https://github.com/js-data/js-data/releases/tag/1.2.1). Please check.
